home-assistant-custom-components.spook: init at 3.0.1

kkoniuszy opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - Tested this with my home-assistant instance
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kkoniuszy committed:
home-assistant-custom-components.spook: init at 3.0.1

kkoniuszy committed:
nixos/home-assistant: fix symlinking multi-manifest custom components

frenck/spook includes a second manifest for an integration. The current
copyCustomComponents script assumed that only one component directory
will be found, which in this case resulted in a malformed symlink
destination:

lrwxrwxrwx 1 hass hass 224 Jun 23 17:23 spook -> '/nix/store/r41ics22zs578avzqf7x86plcgn2q71h-python3.12-frenck-spook-v3.0.1/custom_components/spook/integrations/spook_inverse'$'\n''/nix/store/r41ics22zs578avzqf7x86plcgn2q71h-python3.12-frenck-spook-v3.0.1/custom_components/spook'

kkoniuszy committed:
maintainers: add kkoniuszy

mweinelt commented:
If it ships two components, can we expose them as two dedicated component packages, sharing the same src?

kkoniuszy commented:
> If the repo contains two components, can we expose them as two dedicated component packages, sharing the same src?

@mweinelt They aren't supposed to be installed separately, in HACS there's just a single Spook package. `spook_inverse` is [a helper](https://spook.boo/inverse) that's meant to be a part of `spook`. The separate manifest.json is just a hack that makes it possible to provide a helper – it wouldn't be possible otherwise, as the `integration_type: "hub"` in the core manifest does not let the component provide helpers.

It's just kind of messy. The HACS version does this weird thing where it symlinks `spook_inverse` to `custom_components` then restarts hass, which I patched out here. Separating it into two packages is not the intended use, but it's technically possible as the child component doesn't really depend on any code from the parent. Should I try do that or keep it reasonably close to upstream?

mweinelt commented on pkgs/servers/home-assistant/custom-components/spook/default.nix:
```diff
@@ -0,0 +1,38 @@
+    changelog = "https://github.com/frenck/spook/releases/tag/v${version}";
+    description = "Toolbox for Home Assistant";
+    homepage = "https://spook.boo/";
+    license = "MIT";
```
```suggestion
    license = lib.licenses.mit
```

mweinelt commented on pkgs/servers/home-assistant/custom-components/spook/default.nix:
```diff
@@ -0,0 +1,38 @@
+
+  patches = [./remove-sub-integration-symlink-hack.patch];
+
+  propagatedBuildInputs = [
```
```suggestion
  dependencies = [
```

Newer lingo with the same effect.

mweinelt committed:
nixos/tests/home-assistant: test multi-component packages

The spook package includes two separate integrations and the module was
adapted to account for that scenario. Add a test to ensure the changed
keeps working correctly going forward.

mweinelt commented:
@ofborg test home-assistant

mweinelt commented:
Thanks!
